File: adaptive-arp-api-csharp/Adaptive.Arp.Api/Adaptive/Arp/Api/AttachmentData.cs
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     *
 *
 * =====================================================================================================================
 */

using Sharpen;

namespace Adaptive.Arp.Api
{
	/// <summary>Structure representing the binary attachment data.</summary>
	/// <remarks>Structure representing the binary attachment data.</remarks>
	public class AttachmentData
	{
		/// <summary>The raw data for the current file attachment (byte array)</summary>
		/// <since>ARP1.0</since>
		private byte[] data;

		/// <summary>The data size (in bytes) of the current file attachment</summary>
		/// <since>ARP1.0</since>
		private long dataSize;

		/// <summary>The name of the current file attachment</summary>
		/// <since>ARP1.0</since>
		private string fileName;

		/// <summary>The mime type of the current attachment</summary>
		/// <since>ARP1.0</since>
		private string mimeType;

		/// <summary>The relative path where the contents for the attachment file could be located.
		/// 	</summary>
		/// <remarks>The relative path where the contents for the attachment file could be located.
		/// 	</remarks>
		/// <since>ARP1.0</since>
		private string referenceUrl;

		/// <summary>Constructor used by the implementation</summary>
		/// <param name="data">raw data of the file attachment</param>
		/// <param name="dataSize">size of the file attachment</param>
		/// <param name="fileName">name of the file attachment</param>
		/// <param name="mimeType">mime type of the file attachment</param>
		/// <param name="referenceUrl">relative url of the file attachment</param>
		/// <since>ARP1.0</since>
		public AttachmentData(byte[] data, long dataSize, string fileName, string mimeType
			, string referenceUrl)
		{
			this.data = data;
			this.dataSize = dataSize;
			this.fileName = fileName;
			this.mimeType = mimeType;
			this.referenceUrl = referenceUrl;
		}

		/// <summary>Returns the raw data in byte[]</summary>
		/// <returns>data Octet-binary content of the attachment payload.</returns>
		/// <since>ARP1.0</since>
		public virtual byte[] GetData()
		{
			return data;
		}

		/// <summary>Set the data of the attachment as a byte[]</summary>
		/// <param name="data">Sets the octet-binary content of the attachment.</param>
		/// <since>ARP1.0</since>
		public virtual void SetData(byte[] data)
		{
			this.data = data;
		}

		/// <summary>Returns the size of the attachment as a long</summary>
		/// <returns>dataSize Length in bytes of the octet-binary content.</returns>
		/// <since>ARP1.0</since>
		public virtual long GetDataSize()
		{
			return dataSize;
		}

		/// <summary>Set the size of the attachment as a long</summary>
		/// <param name="dataSize">Length in bytes of the octet-binary content ( should be same as data array length.)
		/// 	</param>
		/// <since>ARP1.0</since>
		public virtual void SetDataSize(long dataSize)
		{
			this.dataSize = dataSize;
		}

		/// <summary>Returns the filename of the attachment</summary>
		/// <returns>fileName Name of the attachment.</returns>
		/// <since>ARP1.0</since>
		public virtual string GetFileName()
		{
			return fileName;
		}

		/// <summary>Set the name of the file attachment</summary>
		/// <param name="fileName">Name of the attachment.</param>
		/// <since>ARP1.0</since>
		public virtual void SetFileName(string fileName)
		{
			this.fileName = fileName;
		}

		/// <summary>Returns the mime type of the attachment</summary>
		/// <returns>mimeType</returns>
		/// <since>ARP1.0</since>
		public virtual string GetMimeType()
		{
			return mimeType;
		}

		/// <summary>Set the mime type of the attachment</summary>
		/// <param name="mimeType">Mime-type of the attachment.</param>
		/// <since>ARP1.0</since>
		public virtual void SetMimeType(string mimeType)
		{
			this.mimeType = mimeType;
		}

		/// <summary>Returns the absolute url of the file attachment</summary>
		/// <returns>referenceUrl Absolute URL of the file attachment for either file:// or http:// access.
		/// 	</returns>
		/// <since>ARP1.0</since>
		public virtual string GetReferenceUrl()
		{
			return referenceUrl;
		}

		/// <summary>Set the absolute url of the attachment</summary>
		/// <param name="referenceUrl">Absolute URL of the file attachment for either file:// or http:// access.
		/// 	</param>
		/// <since>ARP1.0</since>
		public virtual void SetReferenceUrl(string referenceUrl)
		{
			this.referenceUrl = referenceUrl;
		}
	}
}

File: adaptive-arp-api-java/arp-api-specs/src/main/java/me/adaptive/arp/api/AttachmentData.java
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     * Francisco Javier Martin Bueno
 *             <https://github.com/kechis>
 *             <mailto:kechis@gmail.com>
 *
 * =====================================================================================================================
 */

package me.adaptive.arp.api;

/**
 * Structure representing the binary attachment data.
 */
public class AttachmentData {

    /**
     * The raw data for the current file attachment (byte array)
     *
     * @since ARP1.0
     */
    private byte[] data;
    /**
     * The data size (in bytes) of the current file attachment
     *
     * @since ARP1.0
     */
    private long dataSize;
    /**
     * The name of the current file attachment
     *
     * @since ARP1.0
     */
    private String fileName;

    /**
     * The mime type of the current attachment
     *
     * @since ARP1.0
     */
    private String mimeType;

    /**
     * The relative path where the contents for the attachment file could be located.
     *
     * @since ARP1.0
     */
    private String referenceUrl;

    /**
     * Constructor used by the implementation
     *
     * @param data         raw data of the file attachment
     * @param dataSize     size of the file attachment
     * @param fileName     name of the file attachment
     * @param mimeType     mime type of the file attachment
     * @param referenceUrl relative url of the file attachment
     * @since ARP1.0
     */
    public AttachmentData(byte[] data, long dataSize, String fileName, String mimeType, String referenceUrl) {
        this.data = data;
        this.dataSize = dataSize;
        this.fileName = fileName;
        this.mimeType = mimeType;
        this.referenceUrl = referenceUrl;
    }

    public AttachmentData() {
    }

    /**
     * Returns the raw data in byte[]
     *
     * @return data Octet-binary content of the attachment payload.
     * @since ARP1.0
     */
    public byte[] getData() {
        return data;
    }

    /**
     * Set the data of the attachment as a byte[]
     *
     * @param data Sets the octet-binary content of the attachment.
     * @since ARP1.0
     */
    public void setData(byte[] data) {
        this.data = data;
    }

    /**
     * Returns the size of the attachment as a long
     *
     * @return dataSize Length in bytes of the octet-binary content.
     * @since ARP1.0
     */
    public long getDataSize() {
        return dataSize;
    }

    /**
     * Set the size of the attachment as a long
     *
     * @param dataSize Length in bytes of the octet-binary content ( should be same as data array length.)
     * @since ARP1.0
     */
    public void setDataSize(long dataSize) {
        this.dataSize = dataSize;
    }

    /**
     * Returns the filename of the attachment
     *
     * @return fileName Name of the attachment.
     * @since ARP1.0
     */
    public String getFileName() {
        return fileName;
    }

    /**
     * Set the name of the file attachment
     *
     * @param fileName Name of the attachment.
     * @since ARP1.0
     */
    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    /**
     * Returns the mime type of the attachment
     *
     * @return mimeType
     * @since ARP1.0
     */
    public String getMimeType() {
        return mimeType;
    }

    /**
     * Set the mime type of the attachment
     *
     * @param mimeType Mime-type of the attachment.
     * @since ARP1.0
     */
    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    /**
     * Returns the absolute url of the file attachment
     *
     * @return referenceUrl Absolute URL of the file attachment for either file:// or http:// access.
     * @since ARP1.0
     */
    public String getReferenceUrl() {
        return referenceUrl;
    }

    /**
     * Set the absolute url of the attachment
     *
     * @param referenceUrl Absolute URL of the file attachment for either file:// or http:// access.
     * @since ARP1.0
     */
    public void setReferenceUrl(String referenceUrl) {
        this.referenceUrl = referenceUrl;
    }
}
